src/sdk/namespace_fs.js

```diff
@@ -1332,6 +1346,29 @@ class NamespaceFS {
         };
     }
 
+    async _get_version_path(params, fs_context) {
+        if (this.versioning === 'ENABLED' || this.versioning === 'SUSPENDED') {
+            const file_path = this._get_file_path(params);
+            const stat = await nb_native().fs.stat(fs_context, file_path);
+            if (isDirectory(stat)) throw Object.assign(new Error('NoSuchKey'), { code: 'ENOENT' });
+            const xattr_version_id = stat.xattr[XATTR_VERSION_ID];
+            if (!xattr_version_id || !params.version_id || xattr_version_id === params.version_id) {
+                // case 1: when the object was created but versioning was not yet enabled.
+                // Return the latest file path
+                // case 2: the object wanted is the latest version, return the latest file path
+                return file_path;
+            } else {
+                // If the object wanted is among the previous versions, return the
+                // object from $BUCKET_PATH/.versions/version_id
+                const v_path = path.normalize(path.join(this.bucket_path, VERSION_DIR, params.version_id));
```

Comment:
the version path should be of the following form:
```
_get_version_path(key, version_id) {
        const key_version = path.basename(key) + (version_id ? '_' + version_id : '');
        return path.normalize(path.join(this.bucket_path, path.dirname(key), hidden_versions_path, key_version));
    }
```

Reply:
Considered `key_version` to address the file inside `.versions`. Please correct me if wrong